src/client/datascience/insidersNativeNotebookSurveyBanner.ts

```diff
@@ -63,39 +66,43 @@ export class InsidersNativeNotebooksSurveyBanner implements IJupyterExtensionBan
 
     private readonly showBannerState: IPersistentState<ShowBannerWithExpiryTime>;
 
-    private readonly surveyLink: string;
+    private readonly surveyLink: string = 'https://aka.ms/vscjupyternb';
 
     constructor(
         @inject(IApplicationShell) private appShell: IApplicationShell,
         @inject(IPersistentStateFactory) private persistentState: IPersistentStateFactory,
         @inject(IBrowserService) private browserService: IBrowserService,
-        @inject(IVSCodeNotebook) vscodeNotebook: IVSCodeNotebook,
+        @inject(IVSCodeNotebook) private vscodeNotebook: IVSCodeNotebook,
         @inject(IExperimentService) private experimentService: IExperimentService,
         @inject(IApplicationEnvironment) private applicationEnvironment: IApplicationEnvironment,
-        surveyLink = 'https://aka.ms/vscjupyternb'
+        @inject(INotebookExtensibility) private notebookExtensibility: INotebookExtensibility,
+        @inject(IDisposableRegistry) private disposables: IDisposableRegistry
     ) {
-        this.surveyLink = surveyLink;
         this.showBannerState = this.persistentState.createGlobalPersistentState<ShowBannerWithExpiryTime>(
             InsidersNotebookSurveyStateKeys.ShowBanner,
             {
                 data: true
             }
         );
-        vscodeNotebook.onDidOpenNotebookDocument(this.openedNotebook.bind(this));
+    }
+
+    public async activate() {
+        this.vscodeNotebook.onDidOpenNotebookDocument(this.openedNotebook, this, this.disposables);
+        this.notebookExtensibility.onKernelStateChange(this.kernelStateChanged, this, this.disposables);
     }
 
     public async showBanner(): Promise<void> {
         if (this.disabledInCurrentSession) {
             return;
         }
-        // Disable for the current session.
-        this.disabledInCurrentSession = true;
```

Comment:
I think this would've caused the banner to be disabled after the first attempt to show it, so unless the preconditions for showing were met on activation, the banner would not ever show up.